rfd/0158-account-recovery-protections.md

```diff
@@ -0,0 +1,30 @@
+---
+authors: Mike Jensen (mike.jensen@goteleport.com)
+state: draft
+---
+
+# RFD 0158 - Account Recovery Protections
+
+## What
+
+This RFD addresses a flaw from a prior RFD [0029-account-lifecycle.md](0029-account-lifecycle.md) when an account is under a targeted attack.  This RFD clarifies how accounts are recovered when they are being targeted to by an attacker trying to prevent authentication.
+
+## Why
+
+The [Account Lifecycle RFD](0029-account-lifecycle.md) defines how an account can be recovered after the loss of credentials.  However it does not ensure that an account is recoverable under the condition that it is under a targeted attack.
+
+Account authentication can (and should be) locked out after invalid attempts.  Because passwords are user supplied, there are real brute force risks that must be considered.  However this lockout mechanism can also prevent accounts from being accessible by the legitimate user.  We must allow users access to their accounts even under conditions where they are under a targeted attack.
```

Comment:
The password reset mechanism unlocks the account from failed password auth failures.  Which is a common pattern since theoretically the user likely entered an incorrect password repeatedly before starting the password reset flow.